splash-lp-indexer/src/onchain/event.rs

```diff
@@ -320,10 +331,46 @@ pub struct MultiAccountHarvested {
 
 impl<Cx> TryFromLedger<TxViewPartiallyResolved, Cx> for MultiAccountHarvested
 where
-    Cx: Has<DeployedScriptInfo<{ ProtocolValidator::HarvestOrder as u8 }>> + Has<HarvestLimits>,
+    Cx: Has<PermManagerAuthPolicy>
+        + Has<FarmAuthPolicy>
+        + Has<SplashPolicy>
+        + Has<PermManagerAuthPolicy>
+        + Has<HarvestLimits>
+        + Has<BufferWalletScript>
+        + Has<HarvestOrderCreationSlots>
+        + Has<DeployedScriptInfo<{ DaoProtocolValidator::SmartFarm as u8 }>>
+        + Has<DeployedScriptInfo<{ DaoProtocolValidator::PermManager as u8 }>>
+        + Has<DeployedScriptInfo<{ DaoProtocolValidator::HarvestOrder as u8 }>>,
 {
     fn try_from_ledger(repr: &TxViewPartiallyResolved, ctx: &Cx) -> Option<Self> {
-        todo!("DEX-888")
+        let events = OnChainEvents::try_from_ledger(repr, ctx)?;
```

Comment:
This parser already relies on inputs resolved using index data:
```rust
pub struct TxViewPartiallyResolved {
    pub hash: TransactionHash,
    pub inputs: Vec<(TransactionInput, Option<(TransactionOutput, Slot)>)>,
    pub outputs: Vec<TransactionOutput>,
    pub signers: Vec<Ed25519KeyHash>,
    pub slot: u64,
}
```
it's natural to just save Slot in the index along with the corresponding TransactionOutput, then you can use it in parser to retrieve `issued_at` for harvest order

Reply:
It's done, but with struct
```rust
pub struct TimedOutput {
    pub output: TransactionOutput,
    pub slot: Slot,
}
```
due to trait bounds